test/forkMainnet/RFQ.t.sol

```diff
@@ -254,173 +273,257 @@ contract RFQTest is Test, Tokens, BalanceUtil {
-        Snapshot memory makerMakerToken = BalanceSnapshot.take({ owner: offer.maker, token: offer.makerToken });
-        Snapshot memory recTakerToken = BalanceSnapshot.take({ owner: recipient, token: offer.takerToken });
-        Snapshot memory recMakerToken = BalanceSnapshot.take({ owner: recipient, token: offer.makerToken });
-        Snapshot memory feeCollectorBal = BalanceSnapshot.take({ owner: feeCollector, token: offer.makerToken });
+        Snapshot memory makerTakerToken = BalanceSnapshot.take({ owner: rfqOffer.maker, token: Constant.ZERO_ADDRESS });
+        Snapshot memory makerMakerToken = BalanceSnapshot.take({ owner: rfqOffer.maker, token: rfqOffer.makerToken });
+        Snapshot memory recTakerToken = BalanceSnapshot.take({ owner: recipient, token: rfqOffer.takerToken });
+        Snapshot memory recMakerToken = BalanceSnapshot.take({ owner: recipient, token: rfqOffer.makerToken });
+        Snapshot memory feeCollectorBal = BalanceSnapshot.take({ owner: feeCollector, token: rfqOffer.makerToken });
 
-        uint256 fee = (offer.makerTokenAmount * defaultFeeFactor) / Constant.BPS_MAX;
-        uint256 amountAfterFee = offer.makerTokenAmount - fee;
+        uint256 fee = (rfqOffer.makerTokenAmount * defaultFeeFactor) / Constant.BPS_MAX;
+        uint256 amountAfterFee = rfqOffer.makerTokenAmount - fee;
 
-        vm.prank(offer.taker);
-        rfq.fillRFQ(offer, makerSig, defaultPermit, defaultPermit, recipient, defaultFeeFactor);
+        RFQTx memory rfqTx = defaultRFQTx;
+        rfqTx.rfqOffer = rfqOffer;
+        rfqTx.takerRequestAmount = rfqOffer.takerTokenAmount;
 
-        takerTakerToken.assertChange(-int256(offer.takerTokenAmount));
+        vm.prank(rfqOffer.taker, rfqOffer.taker);
+        rfq.fillRFQ(rfqTx, makerSig, defaultPermit, defaultPermit);
+
+        takerTakerToken.assertChange(-int256(rfqOffer.takerTokenAmount));
         takerMakerToken.assertChange(int256(0));
-        makerTakerToken.assertChange(int256(offer.takerTokenAmount));
-        makerMakerToken.assertChange(-int256(offer.makerTokenAmount));
+        makerTakerToken.assertChange(int256(rfqOffer.takerTokenAmount));
+        makerMakerToken.assertChange(-int256(rfqOffer.makerTokenAmount));
         recTakerToken.assertChange(int256(0));
         recMakerToken.assertChange(int256(amountAfterFee));
         feeCollectorBal.assertChange(int256(fee));
     }
 
-    function testCannotFillExpiredRFQOrder() public {
-        vm.warp(defaultOffer.expiry + 1);
+    function testFillWithContract() public {
+        RFQOffer memory rfqOffer = defaultRFQOffer;
+        rfqOffer.flags |= FLG_ALLOW_CONTRACT_SENDER;
+        bytes memory makerSig = _signRFQOffer(makerSignerPrivateKey, rfqOffer);
+
+        RFQTx memory rfqTx = defaultRFQTx;
+        rfqTx.rfqOffer = rfqOffer;
+
+        vm.prank(rfqOffer.taker, makeAddr("anyAddr"));
+        rfq.fillRFQ(rfqTx, makerSig, defaultPermit, defaultPermit);
```

Comment:
nit:
```suggestion
        rfq.fillRFQ(rfqTx, makerSig, defaultPermit, defaultPermit, takerSignature);
```

Reply:
`Fill with contract` is about the `msg.sender` should be a contract. It's not directly relevant that with or without taker signature.

Reply:
Yeah what I mean is that the examples in our tests served as a reference for others to see e.g., how to interact with our contract and so it would be helpful to mimic the realistic behavior. In this scenario `msg.sender` should be a contract and the function it should be calling should the one with taker signature attached.

Reply:
What is the scenario of calling with a contract but trade on an EOA account's behalf? One real example of this test case is a multi-sig wallet, so the taker would be the multi-sig contract.

Reply:
The test case is `testFillWithContract` so it would be a little weird that we prank an EOA as `msg.sender`. The test passes as long we prank different `msg.sender` and `tx.origin` but that's what I meant about mimic realistic behavior - in reality a contract will be the `msg.sender` and in this case it would be calling the `fillRFQ` function with taker signature attached, not the other `fillRFQ` function.

Reply:
Oh I see, my bad.

Reply:
Adjusted in a4f0e46.